File: Leetcode/two_pointers/922_SortArrayByParityII/Solution.py
class Solution:
    def sortArrayByParityII(self, A: List[int]) -> List[int]:
        if not A: return A
        even, odd, length = 0, 1, len(A)
        while even<length and odd<length:
            while even < length and A[even]%2==0:
                even+=2
            while odd < length and A[odd]%2==1:
                odd+=2
            if odd<length and even<length:
                A[odd], A[even] = A[even], A[odd]
        return A

File: Leetcode/two_pointers/922_SortArrayByParityII/Solution.java
class Solution {
    public int[] sortArrayByParityII(int[] A) {
        if(A.length==0)
            return A;
        int odd = 1, even = 0, n = A.length;
        while (even < n && odd < n) {
            while (even < n && A[even] % 2 == 0) {
                even += 2;
            }
            while (odd < n && A[odd] % 2 == 1) {
                odd += 2;
            }
            if (even < n && odd < n)
                swap(A, even, odd);
        }
        return A;
    }
        private void swap(int[] A, int i, int j){
            int temp = A[i];
            A[i] = A[j];
            A[j] = temp;
        }
}
